framework/deproxy_server.py

```diff
@@ -41,13 +43,16 @@ def initiate_send(self):
             self.server.segment_size if self.server.segment_size else deproxy.MAX_MESSAGE_SIZE
         )
 
-        sent = self.socket.send(self.out_buffer[:segment_size])
+        resp = self.response_buffer[self.responses_done]
+        sent = self.socket.send(resp[:segment_size])
+
         if sent < 0:
             return
-        self.out_buffer = self.out_buffer[sent:]
+        self.response_buffer[self.responses_done] = resp[sent:]
```

Comment:
Why you don't use `out_buffer`? If so why `ServerConnection` derived from `asyncore.dispatcher_with_send`?